core/ui_mixins/quick_panel.py

```diff
@@ -112,6 +113,182 @@ def get_arguments(self, selected_action):
+    be called when a branch is selected. If the panel is cancelled, `None`
+    will be passed to `on_done`.
+
+    on_done: a callable
+    ask_remote_first: whether remote should be asked before the branch panel
+            if `False`. the options will be in forms of `remote/branch`
+    selected_branch: if `ask_remote_first`, the selected branch will be
+            `{remote}/{selected_branch}`
+    """
+    bp = BranchPanel(
+        on_done,
+        local_branches_only,
+        remote_branches_only,
+        ignore_current_branch,
+        ask_remote_first,
+        selected_branch)
+    bp.show()
+    return bp
+
+
+class BranchPanel(GitCommand):
+
+    def __init__(
+            self, on_done, local_branches_only=False, remote_branches_only=False,
+            ignore_current_branch=False, ask_remote_first=False, selected_branch=None):
+        self.window = sublime.active_window()
+        self.on_done = on_done
+        self.local_branches_only = local_branches_only
+        self.remote_branches_only = True if ask_remote_first else remote_branches_only
+        self.ignore_current_branch = ignore_current_branch
+        self.ask_remote_first = ask_remote_first
+        self.selected_branch = selected_branch
+
+    def show(self):
+        if self.ask_remote_first:
+            show_remote_panel(
+                lambda remote: sublime.set_timeout_async(
+                    lambda: self.on_remote_selection(remote), 100))
+        else:
+            self.select_branch(remote=None)
+
+    def on_remote_selection(self, remote):
+        if not remote:
+            return
+
+        self.select_branch(remote)
+
+    def select_branch(self, remote=None):
+
+        if self.local_branches_only:
+            self.all_branches = [b.name_with_remote for b in self.get_branches() if not b.remote]
+        elif self.remote_branches_only:
+            self.all_branches = [b.name_with_remote for b in self.get_branches() if b.remote]
+        else:
+            self.all_branches = [b.name_with_remote for b in self.get_branches()]
+
+        if self.ignore_current_branch:
+            current_branch = self.get_current_branch_name()
+            self.all_branches = [b for b in self.all_branches if b != current_branch]
+
+        if remote:
+            self.all_branches = [b for b in self.all_branches if b.startswith(remote + "/")]
+
+        if not self.all_branches:
+            self.window.show_quick_panel(["There are no branches available."], None)
+            return
+
+        self.window.show_quick_panel(
+            self.all_branches,
+            self.on_branch_selection,
+            flags=sublime.MONOSPACE_FONT,
+            selected_index=self.get_pre_selected_branch_index(remote)
+        )
+
+    def get_pre_selected_branch_index(self, remote):
+        pre_selected_index = None
+        if self.selected_branch is None:
+            selected_branch = self.get_current_branch_name()
```

Comment:
Can't we set it to `self.selected_branch` to avoid `self.selected_branch` and `selected_branch`?